### PR TITLE
fix race condition in `(*Spinner).Print`

### PR DESCRIPTION
**Pull request by fortytw2**
Hey! Started using this repo after needing multiple spinners (migrated from `yacspin`) and noticed a race condition caused by `(*Spinner).Print(` when `UpdateMessage(` is used from different goroutines.

I can add a test that reproduces this race detector failure if you'd like, too.

### CODE REVIEW

**Comment by chelnak** [comment]
@fortytw2 Thanks so much for this!

If you wouldn't mind adding a test that would be perfect!

**Comment by fortytw2** [comment]
Of course! Added a test that fails without the additional locks @chelnak. Thanks for the great library :)

**Comment by chelnak** [comment]
Thank you for the test.

It's actually highlighted something unexpected. The race condition actually appears to be coming from `AddSpinner`.

This doesn't happen when you use a mutex inside the method and seems to be validated by your test.

Do you see a similar thing?

File: manager.go
Comment (chelnak) [inline manager.go:67]: Could we `defer` this above the call to append?

File: manager.go
Comment (chelnak) [inline manager.go:37]: Small nit: Could this be renamed to `mutex` so it is consistent with the setup in `spinner.go`?

File: manager.go
Comment (chelnak) [inline manager.go:73]: I'd not considered using this method internally.

It's a fair suggestion and I like it, thank you.

**Review by chelnak — CHANGES_REQUESTED** [review]

**Comment by chelnak** [comment]
@fortytw2 If there is anything I can do to help progress this let me know. 

As soon as we merge, I'm happy to cut a release 🙂

File: manager.go
Comment (fortytw2) [reply manager.go:67]: Done!

File: manager.go
Comment (fortytw2) [reply manager.go:37]: Sure thing - it's pretty standard if there are multiple mutexes to name them like that (after the field they protect) since you want to be as granular as possible when using multiple on a single struct. Totally happy to rename it here though

**Comment by fortytw2** [comment]
Should be good to go @chelnak - let me know if you'd like me to rebase/squash commits in here before merging.

**Comment by chelnak** [comment]
Nah I think we are all good. Your commits are clear enough to me!

**Review by chelnak — APPROVED** [review]
Looks perfect 🤩

**Comment by chelnak** [comment]
@fortytw2 https://github.com/chelnak/ysmrr/releases/tag/v0.2.1 ❤️

**Comment by fortytw2** [comment]
Thanks @chelnak !

**Pull request merged**
